visualization/app/codeCharta/ui/dialog/dialog.download.component.ts

```diff
@@ -47,7 +47,7 @@ export class DialogDownloadController {
 			this.codeMapPreRenderService.getRenderMap(),
 			this.codeMapPreRenderService.getRenderFileMeta(),
 			this.storeService.getState().fileSettings,
-			this._viewModel.fileContent.filter(x => x.isSelected == true).map(x => x.name),
+			this._viewModel.fileContent.filter(x => x.isSelected === true).map(x => x.name),
 			this._viewModel.fileName
```

Comment:
Unfortunately I can't mark all lines of code, but we can combine the filtering and mapping:

```ts
	public download() {
		const downloadSettingsNames: string[]  = []
		for (const fileContent of this._viewModel.fileContent) {
			if (fileContent.isSelected) {
				downloadSettingsNames.push(fileContent.name)
			}
		}

		FileDownloader.downloadCurrentMap(
			this.codeMapPreRenderService.getRenderMap(),
			this.codeMapPreRenderService.getRenderFileMeta(),
			this.storeService.getState().fileSettings,
			downloadSettingsNames,
			this._viewModel.fileName
		)
		this.hide()
	}
```

I'd recommend to move the for-loop to a private function `getSettingsNames` though to keep the one-liner in `download()`

Reply:
I kept it as is, since the number of files to download is low. Thus, the readability wins here IMO.